src/GraphQL/Resolvers/NameFieldResolver.cs

```diff
@@ -20,15 +17,14 @@ public static object Resolve(object source, string name)
                 return null;
             }
 
-            var prop = source.GetType()
-                .GetProperty(name, _flags);
+            var propertyValue = source.GetPropertyValue(name);
 
-            if (prop == null)
+            if (propertyValue == null)
```

Comment:
This change is a little off.  This change would make it impossible to return a null value (which is valid), vs. ensuring that the requested property exists on the object.